theano/tensor/tests/test_basic.py

```diff
@@ -7000,24 +6995,27 @@ def test_infer_shape(self):
         # Flatten
         atens3 = tensor3()
         atens3_val = rand(4, 5, 3)
-        for outdim in (3, 2, 1):
-            self._compile_and_check([atens3],
-                                    [Flatten(outdim)(atens3)],
-                                    [atens3_val], Flatten)
-
-        amat = matrix()
-        amat_val = rand(4, 5)
-        for outdim in (2, 1):
-            self._compile_and_check([amat],
-                                    [Flatten(outdim)(amat)],
-                                    [amat_val], Flatten)
-
-        avec = vector()
-        avec_val = rand(4)
-        outdim = 1
-        self._compile_and_check([avec],
-                                [Flatten(outdim)(avec)],
-                                [avec_val], Flatten)
+        self._compile_and_check([atens3],
+                                [flatten(atens3, 1)],
+                                [atens3_val], Reshape)
+        #for outdim in (3, 2, 1):
```

Comment:
It is related to issue #3420. ```Reshape.infershape``` is badly written and because of this all ```Reshapes``` are not removed from ```atens3.shape``` after compiling.